File: benchmark_phase_c/bench_b3.cs
using System;

class Program
{
    static int DoubleIt(int x) { return x * 2; }

    static Func<int, int> Compose(Func<int, int> f)
    {
        int extra = 10;
        Func<int, int> wrapped = delegate(int x) { return f(x) + extra; };
        return wrapped;
    }

    static int BenchB3(int n)
    {
        var g = Compose(DoubleIt);
        int total = 0;
        int i = 0;
        while (i < n)
        {
            total = total + (g(i) % 1000);
            i = i + 1;
        }
        return total;
    }

    static void Main(string[] args)
    {
        int n = int.Parse(args[0]);
        Console.WriteLine(BenchB3(n));
    }
}

File: benchmark_phase_c/bench_b3.py
import sys


def double_it(x):
    return x * 2


def compose(f):
    extra = 10

    def wrapped(x):
        return f(x) + extra

    return wrapped


def bench_b3(n):
    g = compose(double_it)
    total = 0
    i = 0
    while i < n:
        total = total + (g(i) % 1000)
        i = i + 1
    return total


if __name__ == '__main__':
    n = int(sys.argv[1])
    print(bench_b3(n))
